pkg/controller/garbagecollector/graph_builder.go

```diff
@@ -166,15 +166,30 @@ func (gb *GraphBuilder) controllerFor(resource schema.GroupVersionResource, kind
 			gb.graphChanges.Add(event)
 		},
 	}
-	shared, err := gb.sharedInformers.ForResource(resource)
+	generic, err := gb.sharedInformers.ForResource(resource)
 	if err != nil {
 		klog.V(4).Infof("unable to use a shared informer for resource %q, kind %q: %v", resource.String(), kind.String(), err)
-		return nil, nil, err
+		return nil, err
 	}
 	klog.V(4).Infof("using a shared informer for resource %q, kind %q", resource.String(), kind.String())
 	// need to clone because it's from a shared cache
-	shared.Informer().AddEventHandlerWithResyncPeriod(handlers, ResourceResyncTime)
-	return shared.Informer().GetController(), shared.Informer().GetStore(), nil
+	generic.Informer().AddEventHandlerWithResyncPeriod(handlers, ResourceResyncTime)
+	return generic.Informer(), nil
+}
+
+// filterStoppedResources filters out the resources that have been removed
+// and reinstalled from the cluster in between garbage collector syncs.
+func (gb *GraphBuilder) filterStoppedResources(set resourceSet) {
+	gb.stoppedResourcesLock.Lock()
+	defer gb.stoppedResourcesLock.Unlock()
+	for resource := range gb.stoppedResources {
+		// filter resource from the resource set used by Sync
+		// so that it's deletion is recognized on the first Sync cycle.
+		delete(set, resource)
+	}
+	// reset stoppedResources so that all stopped resources
+	// are re-installed on the next sync cycle.
```

Comment:
Do you mean the informer of stopped resources,  that are re-installed on the same sync cycle, will be re-start on the next sync cycle?

Reply:
Yes, comment updated